x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/normalization/rule_actions.ts

```diff
@@ -107,7 +107,7 @@ export const transformActions = (
   if (alertAction != null && alertAction.length !== 0) {
     return alertAction.map((action) => transformAlertToRuleAction(action));
   } else if (legacyRuleActions != null) {
-    return legacyRuleActions.actions;
+    return legacyRuleActions.actions as RuleResponse['actions'];
```

Comment:
Perhaps we can change the return type of this function to `RuleResponse['actions'] | LegacyRuleResponse['actions']` to correctly represent the return type rather than casting here?